packages/cms-base/components/public/cms/element/CmsElementProductListing.vue

```diff
@@ -35,19 +41,44 @@ const {
 
 const route = useRoute();
 const router = useRouter();
+const limit = ref(
+  route.query.limit
+    ? Number(route.query.limit)
+    : props.content?.data?.listing?.limit
+      ? Number(props.content?.data?.listing?.limit)
+      : 15,
+);
 
 const changePage = async (page: number) => {
   await router.push({
     query: {
       ...route.query,
       p: page,
+      limit: limit.value,
     },
   });
-  changeCurrentPage(
+  await changeCurrentPage(
     page,
     route.query as unknown as RequestParameters<"searchPage">,
   );
 };
+
+const changeLimit = async (limit: Event) => {
+  const select = limit.target as HTMLSelectElement;
+
+  await router.push({
+    query: {
+      ...route.query,
+      limit: select.value,
+      p: 1,
+    },
+  });
+  await changeCurrentPage(
+    1,
+    route.query as unknown as RequestParameters<"searchPage">,
+  );
+};
+
 const isProductListing = computed(
   () => props.content?.type === "product-listing",
 );
```

Comment:
maybe we'd compare initialListing with currentListing (router one) and if there is a mismatch invoke search? That would at leat be decent workaround for Vercel query params cache issue. WDYT?

Reply:
Implemented some ugly workaround. :sob: